CoreVisualSchema/src/au/gov/asd/tac/constellation/graph/schema/visual/VisualSchemaFactory.java

```diff
@@ -297,7 +300,44 @@ public void completeTransaction(final GraphWriteMethods graph, final int transac
         }
 
         private ConstellationColor randomColor() {
-            return ConstellationColor.getColorValue(random.nextFloat(), random.nextFloat(), random.nextFloat(), 1.0F);
+            final Preferences prefs = NbPreferences.forModule(ApplicationPreferenceKeys.class);
+            //Retrieve colorblind mode selection preference 
+            final String colorMode = prefs.get(ApplicationPreferenceKeys.COLORBLIND_MODE, ApplicationPreferenceKeys.COLORBLIND_MODE_DEFAULT);
+
+            final float brightenFloat = 0.10F; //Value to inflate randomly generated low floats 
+            final float lowFloat = 0.10F;
+            float randFloat1 = random.nextFloat();
+            float randFloat2 = random.nextFloat();
+            float randFloat3 = random.nextFloat();
+
+            //Change node color randomiser based on colorblind mode selection
+            switch (colorMode) {
+                case "None":
+                    return ConstellationColor.getColorValue(randFloat1, randFloat2, randFloat3, 1.0F);
+                case "Deuteranopia":
+                    //Ensure randomised color does not generate an RGB value which is too dark for user contrast.
+                    if (randFloat1 <= lowFloat && randFloat3 <= lowFloat) {
+                        randFloat1 += brightenFloat;
+                        randFloat3 += brightenFloat;
+                    }
+
+                    return ConstellationColor.getColorValue(randFloat1, 0F, randFloat3, 1.0F);
+                case "Protanopia":
+                    if (randFloat1 <= lowFloat && randFloat3 <= lowFloat) {
+                        randFloat3 += brightenFloat;
+                        randFloat1 += brightenFloat;
+                    }
+                    return ConstellationColor.getColorValue(randFloat1, 0F, randFloat3, 1.0F);
```

Comment:
Both these cases are essentially running the same code ... is this correct ?
If so, the code can be simplified by stripping out lines 318 to 324,
so the two cases go into the same code block.